lib/protractor.ts

```diff
@@ -0,0 +1,1208 @@
+// Util from NodeJs
+import * as util from 'util';
+import * as url from 'url';
+import * as net from 'net';
+import * as helper from './util';
+import {Logger} from './logger2';
+import {ElementArrayFinder, ElementFinder, build$, build$$} from './element';
+let Plugins = require('./plugins');
+
+let clientSideScripts = require('./clientsidescripts');
+let ProtractorBy = require('./locators').ProtractorBy;
+import {ExpectedConditions} from './expectedConditions';
+
+let webdriver = require('selenium-webdriver');
+let Command = require('selenium-webdriver/lib/command').Command;
+let CommandName = require('selenium-webdriver/lib/command').Name;
+
+// jshint browser: true
+/* global angular */
+
+const DEFER_LABEL = 'NG_DEFER_BOOTSTRAP!';
+const DEFAULT_RESET_URL = 'data:text/html,<html></html>';
+const DEFAULT_GET_PAGE_TIMEOUT = 10000;
+
+let logger = new Logger('protractor');
+
+/*
+ * Mix in other webdriver functionality to be accessible via protractor.
+ */
+// TODO: verify this iterates over all the webdriver methods with TypeScript.
+for (var foo in webdriver) {
+  exports[foo] = webdriver[foo];
+}
+
+/**
+ * @type {ProtractorBy}
+ */
+export let By = new ProtractorBy();
+
+/**
+ * @type {ElementFinder}
+ */
+export {ElementFinder} from './element';
```

Comment:
I think we can do `export {ElementFinder, ElementArrayFinder} from './element';`